at_client/lib/src/service/sync_service_impl.dart

```diff
@@ -491,28 +494,37 @@ class SyncServiceImpl implements SyncService, AtSignChangeListener {
         var metaData =
             await _atClient.getLocalSecondary()!.keyStore!.getMeta(key);
         if (metaData != null) {
-          key = '$key$_metadataToString(metaData)';
+          key = '$key${_metadataToString(metaData)}';
         }
         command = 'update:meta:$key';
         break;
       case CommitOp.UPDATE_ALL:
         var key = entry.atKey;
-        var value = await _atClient.getLocalSecondary()!.keyStore!.get(key);
-        var metaData =
-            await _atClient.getLocalSecondary()!.keyStore!.getMeta(key);
+        AtData value = await _atClient.getLocalSecondary()!.keyStore!.get(key);
         var keyGen = '';
-        if (metaData != null) {
-          keyGen = _metadataToString(metaData);
+        // If metadata.encoding is not empty, the value is encoded.
+        // When value is encoded, use update:json variant in update regex to
+        // send command to cloud secondary
+        if (value.metaData != null && value.metaData?.encoding != null) {
```

Comment:
Should this be `value.metaData!.encoding`?

Reply:
As a part of reverting JSON approach, removed the above lines.